packages/amplify-migration-e2e/src/templategen.ts

```diff
@@ -118,25 +120,53 @@ async function assertStepCompletion(command: string) {
 
 async function assertRefactorStepCompletion(command: string) {
   const processResult = JSON.parse(await executeCommand(command));
-  return processResult.Status;
+  return processResult;
 }
 
-export async function stackRefactor(projRoot: string, category: RefactorCategory, bucketName: string) {
-  const { gen1ResourceIds, gen1ResourceDetails } = await getGen1ResourceDetails(projRoot, category);
-
-  const readmeFilePath = path.join(projRoot, '.amplify', 'migration', 'templates', category, 'MIGRATION_README.md');
-  const readmeContent = fs.readFileSync(readmeFilePath, 'utf-8');
-  const { step1Commands, step2commands, step3Commands } = getCommandsFromReadme(readmeContent);
+async function takeTemplateSnapshot(projectRoot: string, category: RefactorCategory, templateName: string) {
+  const templateFilePath = path.join(projectRoot, '.amplify', 'migration', 'templates', category, templateName);
+  const templateFileContent = readJsonFile(templateFilePath);
+  return templateFileContent;
+}
 
+export async function executeStackRefactorSteps(projRoot: string, category: RefactorCategory, bucketName: string) {
+  const readmeContent = readMigrationReadmeFile(projRoot, category);
+  const { step1Commands, step2commands, step3Commands } = getStackRefactorCommandsFromReadme(readmeContent);
   await executeStep1(projRoot, step1Commands);
   await executeStep2(projRoot, step2commands);
   await executeStep3(projRoot, step3Commands, bucketName);
+}
+
+export async function stackRefactor(projRoot: string, projName: string, category: RefactorCategory, bucketName: string) {
+  const { gen1ResourceIds, gen1ResourceDetails } = await getGen1ResourceDetails(projRoot, category);
+  removeProperties(gen1ResourceDetails, ['CorsConfiguration.CorsRules[0].Id', 'Tags']);
```

Comment:
is this needed since gen2 codegen adds tags to CORS ? If so, can we add a comment stating it.

Reply:
Yeah `Tags` and `Id` in `CorsRules` are modified when `npx ampx sandbox` is run. So, we would have to remove those fields before asserting that the source resource is same before and after stack refactor.

Reply:
Done. [bd7aa34](https://github.com/aws-amplify/amplify-cli/pull/14009/commits/bd7aa34ecd6f607bb1f9a5892b3cda4fe6a1ce43)